### PR TITLE
storing workflows as datasets

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[UTM-Sentinel-2 Dataset ausprobieren](https://trello.com/c/VU2iMMFE/375-utm-sentinel-2-dataset-ausprobieren)

File: geoengine/types.py [hunk 1]
```diff
@@ -138,13 +138,47 @@ def srs(self) -> str:
 
         return self.__srs
 
+    def __dict__(self) -> Dict[str, Any]:
```
Comment (michaelmattig) [inline geoengine/types.py:141]: we would need to distinguish between raster and vector query rectangle

File: geoengine/types.py [hunk 2]
```diff
@@ -138,13 +138,47 @@ def srs(self) -> str:
...
+        }
+
 
 class ResultDescriptor:  # pylint: disable=too-few-public-methods
     '''
     Base class for result descriptors
     '''
 
-    @staticmethod
+    @ staticmethod
```
Comment (michaelmattig) [inline geoengine/types.py:181]: is this the correct formatting?

File: geoengine/workflow.py [hunk 3]
```diff
@@ -425,6 +425,39 @@ def get_provenance(self) -> List[ProvenanceOutput]:
 
         return [ProvenanceOutput.from_response(item) for item in response]
 
+    def save_as_layer(self, name: str, description: str, bbox: QueryRectangle) -> InternalDatasetId:
```
Comment (michaelmattig) [inline geoengine/workflow.py:428]: maybe change order and give description a default value of empty string?

Comment (michaelmattig) [reply geoengine/workflow.py:428]: also we should name the method `save_as_dataset` or just `save`

File: tests/test_workflow_storage.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import unittest
+
+import requests_mock
+
+from geoengine.types import InternalDatasetId, QueryRectangle
+import geoengine as ge
+
+
+class WorkflowStorageTests(unittest.TestCase):
+    '''WMS test runner'''
```
Comment (michaelmattig) [inline tests/test_workflow_storage.py:13]: copy/pasta

Comment (ChristianBeilschmidt) [reply tests/test_workflow_storage.py]: fbe1682

File: geoengine/types.py (hunk 2 above)
Comment (ChristianBeilschmidt) [reply geoengine/types.py]: fbe1682

File: geoengine/types.py (hunk 1 above)
Comment (ChristianBeilschmidt) [reply geoengine/types.py:141]: 24f3e00

File: geoengine/workflow.py (hunk 3 above)
Comment (ChristianBeilschmidt) [reply geoengine/workflow.py]: a2b883c

File: geoengine/types.py (hunk 2 above)
Comment (ChristianBeilschmidt) [reply geoengine/types.py]: f6853ba

**Review by michaelmattig — APPROVED** [review]

**Pull request merged**